### PR TITLE
Alternative form of detection and ground truth grouping.

### PR DESCRIPTION
**Pull request by mattbrown11**

### CODE REVIEW

File: angel_system/eval/ptg_evaluate.py [hunk 1]
```diff
@@ -91,55 +91,115 @@ def run_eval(args):
...
+
+    gt_true_mask = np.zeros((len(time_windows), len(labels)), dtype=bool)
+    for i in range(len(gt)):
+        ind1, ind2 = get_time_wind_range(gt['start'][i], gt['end'][i])
+        correct_label = gt['class'][i].strip().rstrip('.')
+        correct_class_idx = labels.index(correct_label)
+
+        #print(gt['start'][i], gt['end'][i], ind1, ind2, correct_class_idx)
+
+        if np.any(gt_true_mask[ind1:ind2, :]):
```
Comment (mattbrown11) [inline angel_system/eval/ptg_evaluate.py]: take this out

File: angel_system/eval/ptg_evaluate.py [hunk 2]
```diff
@@ -91,55 +91,115 @@ def run_eval(args):
...
+            raise Exception()
+
+        gt_true_mask[ind1:ind2, correct_class_idx] = True
+
+    if not np.all(np.sum(gt_true_mask, axis=1) <= 1):
+        raise Exception('Conflicting ground truth for same time windows')
+
+
```
Comment (mattbrown11) [inline angel_system/eval/ptg_evaluate.py:161]: Add check that if a row isn't `gt_true_mask` isn't populated, populate 'background' label to True.

File: angel_system/eval/ptg_evaluate.py [hunk 3]
```diff
@@ -91,55 +91,115 @@ def run_eval(args):
...
+    # Any time the ground truth class changes, we want to add in uncertainty
+    # padding, but there should always be at least one time window at the
+    # center of the ground-truth span.
+    pad = int(np.round(args.uncertain_pad/dt))
+    if pad > 0:
+        ind = np.where(np.any(np.diff(gt_true_mask, axis=0), axis=1))[0]
```
Comment (mattbrown11) [inline angel_system/eval/ptg_evaluate.py]: Make sure the indexing here makes sense.

**Comment by hdefazio** [comment]
LGTM

**Pull request merged**